File: py_payment_api/app/services/store/payment/subscription/subscription_collection_service.py
from ..abstract.item_collection_service import ItemCollectionService
from ....store.enum import ItemType

from typing import Generic, TypeVar
from enum import StrEnum
ITEM_CATEGORY = TypeVar('ITEM_CATEGORY', bound=StrEnum)
class SubscriptionCollectionService(Generic[ITEM_CATEGORY], ItemCollectionService[ITEM_CATEGORY]):
    def __init__(self) -> None:
        super().__init__(ItemType.SUBSCRIPTION, ItemType.SUBSCRIPTION)

File: tsPaymentApi/src/services/store/payment/subscription/SubscriptionCollectionService.ts
import { ItemCollectionService } from '../abstract/ItemCollectionService';
import { ItemType } from '../../../../types/store';

/**
 * Subscription collection service that matches the Python implementation.
 * Inherits all functionality from ItemCollectionService and only initializes
 * with ItemType.SUBSCRIPTION.
 */
export class SubscriptionCollectionService<TCategory extends string> extends ItemCollectionService<TCategory> {
    constructor() {
        super(ItemType.SUBSCRIPTION, ItemType.SUBSCRIPTION);
    }
} 